setup.ps1

```diff
@@ -0,0 +1,229 @@
+    $openLog = Read-Host
+    if ($openLog -eq 'Y' -or $openLog -eq 'y') {
+      Start-Process notepad.exe -ArgumentList $LogFilePath
+    }
+  }
+
+  if ($waitForKeypress) {
+    Write-Log "Press any key to exit..."
+    $host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown") | Out-Null
+  }
+
+  return $exitCode
+}
+
+# Function to handle installation
+function Install {
+  param ([string]$InputPath)
+  
+  if (-not $InputPath) {
+    Write-Log "No input provided for installation." -Level 'ERROR'
+    Exit-Script -exitCode 1
+  }
+
+  Write-Log "Installing $InputPath..."
+  $installCmd = if (Test-Path $InputPath) { $VenvPip + @('install', '-r', $InputPath) } else { $VenvPip + @('install', $InputPath) }
+  $output = & $installCmd[0] $installCmd[1..$installCmd.Length] 2>&1
+  $output | Out-File $LogFilePath -Append
+  if ($LASTEXITCODE -ne 0) {
+    Write-Log "Conflict detected. Exiting now..." -Level 'ERROR'
+    Exit-Script -exitCode 1
+  }
+}
+
+# Function to get the installed Python version tag for wheel compatibility
+function Get-PythonVersionTag {
+  $pythonVersion = & python -c "import sys; print(f'cp{sys.version_info.major}{sys.version_info.minor}')"
+  $architecture = & python -c "import platform; print('win_amd64' if platform.machine().endswith('64') else 'win32')"
+  return "$pythonVersion-$architecture"
+}
+
+# Exit in test environment
+if ($MyInvocation.InvocationName -ne $MyInvocation.MyCommand.Name) {
+  exit
+}
+
+# Check for admin privileges and elevate if necessary
+if (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
+  Write-Log "Requesting administrative privileges..." -Level 'ERROR'
+  Start-Process PowerShell -ArgumentList "-File `"$PSCommandPath`"" -Verb RunAs
+  Exit-Script -exitCode 1 -isSubShell $false
+}
+
+# Log file setup
+"Admin rights confirmed" | Out-File $LogFilePath -Append
+"Starting the script operations..." | Out-File $LogFilePath -Append
+
+# Navigate to the project directory
+Set-Location -Path $ProjectDir
+"Current directory: $(Get-Location)" | Out-File $LogFilePath -Append
+
+# Define the path to the Python executable in the virtual environment
+$VenvPython = Join-Path $ProjectDir "$VenvName\Scripts\python.exe"
+
+# Check if the virtual environment exists, if not, create it
+if (-Not (Test-Path $VenvPython)) {
+  Write-Log "Virtual environment not found. Creating virtual environment..." -IsError $false
+  python -m venv "$VenvName"
+  if (-Not (Test-Path $VenvPython)) {
+    Write-Log "Failed to create virtual environment." -Level 'ERROR'
+    Exit-Script -exitCode 1
+  }
+  Write-Log "Virtual environment created successfully." -IsError $false
+}
+
+# Define the pip command using the Python executable
+$VenvPip = @($VenvPython, '-m', 'pip')
+
+# Activate the virtual environment
+$OldVirtualPath = $env:PATH
+$env:PATH = "$ProjectDir\$VenvName\Scripts;$env:PATH"
```

Comment:
That's why it restores the env on exit. It also doesn't require extended privileges. On StackOverflow they mention running the ps1 script for activating venv requires the user to adjust the scope. I don't know if this is necessary for the current script already, but if this is not the case, maybe the current solution is better because it avoids to set the extended scope. https://stackoverflow.com/questions/18713086/virtualenv-wont-activate-on-windows